lib/lighthouse/benefits_documents/configuration.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+require 'common/client/configuration/rest'
+require 'lighthouse/auth/client_credentials/jwt_generator'
+require 'lighthouse/auth/client_credentials/service'
+
+module BenefitsDocuments
+  ##
+  # HTTP client configuration for the {BenefitsClaims::Service},
+  # sets the base path, the base request headers, and a service name for breakers and metrics.
+  #
+  class Configuration < Common::Client::Configuration::REST
+    self.read_timeout = Settings.lighthouse.benefits_documents.timeout || 20
+
+    SYSTEM_NAME = 'vets.gov'
+    API_SCOPES = %w[system/documents.read system/documents.write].freeze
+    DOCUMENTS_PATH = 'benefits-documents/v1/documents'
+    TOKEN_PATH = 'oauth2/benefits-documents/system/v1/token'
+
+    ##
+    # @return [Config::Options] Settings for benefits_claims API.
+    #
+    def documents_settings
+      Settings.lighthouse.benefits_settings
+    end
+
+    def global_settings
```

Comment:
There's a push to reuse the same auth between all of the services, so I broke out settings to `global` and `#{api_name}_settings`. Other services don't do this, but I would imagine this is how it will need to be handled

Reply:
What do you think about moving the settings you're relying on from `benefit_claims` to live at a higher level to avoid that coupling? Something like:

```yml
# Settings.yml
lighthouse:
  auth:
    ccg:
      client_id: ~
      rsa_key: ~
  benefit_claims:
    blah:
  letters:
     blah:
```

Of course, we'd have to add the param names and values in the parameter store.

Reply:
I'm happy to move it, I just wasn't sure if lighthouse was ready to create a global auth like that. I haven't added anything to the parameter store before either. Is there documentation for how to do that?

Reply:
Updated

Reply:
Sorry, yes! [There are docs on adding and configuring a secret](https://depo-platform-documentation.scrollhelp.site/developer-docs/vets-api-on-eks#VetsAPIonEKS-SecretValues)

Reply:
Let's also get @jerekshoe's opinion, since he did a lot of work on the CCG stuff. Jerek, what do you thing - should we do this or hold off, or is there another way you know of?

Reply:
I'm fine with doing it that way, I would just copy the existing parameters for benefits claims and use the new ones here. We can migrate the others in the future

Reply:
The reason we wanted to use the benefits variables for now is it would allow us to get this in staging and test it without making devops repo or parameter store changes and we should be using the exact same values.

Reply:
and if they ever change they should change for both as long as LH says we should use the id and rsa key